lib/api/core/models/repositories/profileRepository.js

```diff
@@ -124,30 +119,25 @@ module.exports = function (kuzzle) {
     }
 
     // force "default" role if the profile does not have any role in it
-    if (!profile.roles || profile.roles.length === 0) {
-      profile.roles = [ {_id: 'default'} ];
+    if (!profile.policies || profile.policies.length === 0) {
+      profile.policies = [ {_id: 'default'} ];
     }
 
-    q(_.assignIn(profile, data))
-      .then(result => {
-        o = result;
-        return kuzzle.repositories.role.loadRoles(extractRoleIds(result.roles));
-      })
-      .then((roles) => {
+    _.assignIn(profile, data);
+
+    return kuzzle.repositories.role.loadRoles(extractRoleIds(profile.policies))
+      .then(roles => {
         var
-          rolesNotFound = _.difference(extractRoleIds(o.roles), extractRoleIds(roles));
+          rolesNotFound = _.difference(extractRoleIds(profile.policies), extractRoleIds(roles));
 
         // Fail if not all roles are found
         if (rolesNotFound.length) {
-          deferred.reject(new NotFoundError(`Unable to hydrate the profile ${data._id}. The following roles don't exist: ${rolesNotFound}`));
-          return deferred.promise;
+          return q.reject(new NotFoundError(`Unable to hydrate the profile ${data._id}. The following roles don't exist: ${rolesNotFound}`));
         }
 
-        deferred.resolve(o);
+        return profile;
       })
-      .catch(error => deferred.reject(error));
-
-    return deferred.promise;
+      .catch(error => q.reject(error));
```

Comment:
nitpicking: this line is superfluous, since the whole promise chain is returned to the caller anyway, including rejections.